Make libvirt module work under Fedora

larsks opened the pull request:
This adds basic support for Fedora to the 'libvirt' module. It's mostly just a block in params.pp, but it also adds a user {} resource to create the 'libvirt' user.

Commit:
make libvirt module work under Fedora

Commit:
make libvirt a system user

kbarber commented:
Thanks! I'll make some comments on the submission.

Can you please raise a ticket on this here:

http://projects.puppetlabs.com/projects/modules/issues/new

And sign the CLA as well:

https://projects.puppetlabs.com/contributor_licenses/sign

We'll need that to accept any code submission.

larsks commented:
> Can you please raise a ticket on this here:
> 
> http://projects.puppetlabs.com/projects/modules/issues/new
> 
> And sign the CLA as well:
> 
> https://projects.puppetlabs.com/contributor_licenses/sign
> 
> We'll need that to accept any code submission.

Done and Done.

Cheers,

-- Lars

kbarber commented on manifests/init.pp:
```diff
@@ -87,6 +87,15 @@ class libvirt (
+  # Users and groups. #
+  #####################
+
+  user { $user:
```
This is potentially troublesome. The package normally creates the user doesn't it? If this was to occur before the package you would be creating the user with a different uid.

Are you doing this to work around a specific problem?

larsks replied:
> This is potentially troublesome. The package normally creates the user doesn't it?

The 'libvirt' package for Fedora creates neither a libvirt user or
group.  We could probably get away with just creating the group,
rather than both, since currently you're not using $user for anything.

There are two options for controlling access:
- Follow the model used by the module for Debian/Ubuntu.  Create the
  corresponding group, set the various auth_\* options to 'none', and use
  Unix socket permissions to control access.
- Use PolicyKit.  This still requires a system group (well, not
  really, but that makes the most sense), but instead of using Unix
  socket permissions it requires an entry under /etc/polkit-1.
  PolicyKit offers somewhat more granular control (you could, say, offer
  password-free access to console users, and require password
  authentication for anyone else), but for the basic use case there's
  not a clear advantage.  The PolicyKit stuff is documented briefly at
  http://libvirt.org/auth.html.

I'm inclined to stick with the first option, since that means the
model is the same for both Fedora and Ubuntu/Debian.

kbarber replied:
Lol. I'm not even using the user - just the group :-). We can probably remove the user param completely.

I'd work with just the group. Also - you want to 'require' the package so this doesn't go and create the group before the package does - this will matter for Debian/Ubuntu.

kbarber commented:
http://projects.puppetlabs.com/issues/10822

kbarber commented:
Also - you might want to update the README to mention this was tested on Fedora :-). It only states Debian Wheezy.

Commit:
only create group, not user

Commit:
regenerate config files if they are missing

This makes sure that libvirtd.conf and qemu.conf are regenerated
if they have been removed.

Commit:
Updated README to reflect testing on Fedora

Commit:
require package per suggestion from Keb Barber

Commit:
made centos work like fedora

belmiromoreira commented:
Any update for the Fedora support? I would like to try it in SLC.

ccaum commented:
@kbarber Where are at on this?

belmiromoreira commented:
I tried this module with the new commits for Fedora/CentOS and it works fine on SLC (of course I needed to add SLC as a supported OS).
It would be great if the support for the Redhat family is integrated.

puppetcla commented:
CLA Signed by larsks on 2011-11-13 21:00:00 -0800
